pkg/network/setup/configstate.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+)
+
+type ConfigState struct {
+	cacheCreator cacheCreator
+	vmiUID       string
+}
+
+func NewConfigState(cc cacheCreator, vmiUID string) ConfigState {
+	return ConfigState{cc, vmiUID}
+}
+
+func (c ConfigState) RunOnce(networkName string, prepareFunc func() error, configFunc func() error) error {
+	state, err := c.read(networkName)
+	if err != nil {
+		return err
+	}
+
+	switch state {
+	case cache.PodIfaceNetworkPreparationStarted:
+		return neterrors.CreateCriticalNetworkError(
+			fmt.Errorf("pod interface %s network preparation cannot be restarted", networkName),
+		)
+	case cache.PodIfaceNetworkPreparationFinished:
+		return nil
+	}
+
+	if ferr := prepareFunc(); ferr != nil {
+		return ferr
+	}
+
+	if werr := c.write(networkName, cache.PodIfaceNetworkPreparationStarted); werr != nil {
+		return fmt.Errorf("failed to mark configuration as started for %s: %v", networkName, werr)
+	}
+
+	// The configuration function must be called *after* the preparation functions
+	// since it mutates the pod interface from which those generator methods get their info from.
```

Comment:
The comment has to be adapted,  the `RunOnce` is not aware to the content of the `prepareFunc` therefore mentioning  the `generator`s in the comments is wrong imo.

Reply:
Done